src/Peachpie.CodeAnalysis/FlowAnalysis/Passes/TransformationRewriter.CopyAnalysis.cs

```diff
@@ -0,0 +1,352 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Pchp.CodeAnalysis.Semantics;
+using Pchp.CodeAnalysis.Semantics.Graph;
+using Pchp.CodeAnalysis.Symbols;
+using Peachpie.CodeAnalysis.Utilities;
+
+namespace Pchp.CodeAnalysis.FlowAnalysis.Passes
+{
+    partial class TransformationRewriter
+    {
+        /// <summary>
+        /// Each state in copy analysis maps each variable to a set of <see cref="BoundCopyValue"/> instances.
+        /// Whenever a variable is modified, all the mapped copy operations are considered unavailable for removal
+        /// due to the possible aliasing.
+        /// </summary>
+        private struct CopyAnalysisState
+        {
+            private BitMask[] _data;
```

Comment:
The shortest name sufficiently describing the semantics I could think of was `varToCopySetMap`, which sounds a bit cumbersome. It's the only field in the structure and its semantics is described in the structure summary, so I thought it might be enough. Do you have any suggestions for a better name?

Reply:
maybe `_state` or `_varState`

Reply:
OK, in fc85bbee80464887d9a9d9d8e4f10b5887f079c3.